fuzz/ParseFuzzer.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright 2020 Google LLC
+// Copyright 2025 Mikhail Khachaiants <mkhachaiants@gmail.com>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "uriparser/Uri.h"
+#include "uriparser/UriIp4.h"
+#include "FuzzingUtils.h"
+#include <cstddef>
+#include <cstring>
+#include <vector>
+
+
+
+class UriHolder {
+public:
+	UriHolder() {
+		memset((void *)&uri_, 0, sizeof(uri_));
+	}
+
+	~UriHolder() {
+		URI_FUNC(FreeUriMembers)(&uri_);
+	}
+
+	URI_TYPE(Uri) * get() {
+		return &uri_;
+	}
+
+private:
+	URI_TYPE(Uri) uri_;
+};
+
+
+
+void Escapes(const String & uri) {
+	const URI_CHAR * first = uri.c_str();
+	// A new line char takes 6 char to encode.
```

Comment:
This is about "%0D%0A" with `normalizeBreaks` enabled, right? Would be cool to enrich this comment a bit to address the natural questions coming up. :pray:

Reply:
I think yes, the comment is extended